velox/functions/prestosql/StringFunctions.cpp

```diff
@@ -434,6 +434,130 @@ class Replace : public exec::VectorFunction {
     return {{0, 2}};
   }
 };
+
+/**
+ * replace_first(string, search, replace) -> varchar
+ * Replaces the first instances of ``search`` with ``replace`` in ``string``.
+ * If search is an empty string, it inserts replace at the beginning of the
+ * string.
+ **/
+class ReplaceFirst : public exec::VectorFunction {
```

Comment:
Can you also de-duplicate the code for ReplaceFirst and reuse the one for replace?

You can do this by changing the constructor of replace to take a bool like `Replace(bool isReplaceFirst)` then using this wherever the implementation diverges. Also, add the same bool in `signatures(bool isReplaceFirst)` and only register the signature relevant to it. Finally, while registering you can pass in true and false corresponding to ReplaceFirst and Replace respectively. For a code reference on how the same vector function is used for different UDFs, you can checkout ArraySort.cpp

Reply:
I can add it. I think sigantures would be a no-op, right? replace anyways needs 
```
        exec::FunctionSignatureBuilder()
            .returnType("varchar")
            .argumentType("varchar")
            .argumentType("varchar")
            .argumentType("varchar")
            .build(),
    ```
    which is what replaceFirst needs

Reply:
nvm I misunderstood, i got it, we need to apply the delta during registration. thank you!